addons/html_builder/static/src/website_builder/plugins/options/google_maps_option/google_maps_option_plugin.js

```diff
@@ -0,0 +1,355 @@
+        this.gpsMapCache = new Map();
+    }
+
+    destroy() {
+        // Without this, the google library injects elements inside the backend
+        // DOM but do not remove them once the editor is left. Notice that
+        // this is also done when the widget is destroyed for another reason
+        // than leaving the editor, but if the google API needs that container
+        // again afterwards, it will simply recreate it.
+        for (const container of document.body.querySelectorAll('.pac-container')) {
+            container.remove();
+        }
+    }
+
+    getActions() {
+        return {
+            resetMapColor: {
+                apply: ({ editingElement }) => {
+                    editingElement.dataset.mapColor = "";
+                },
+            },
+            showDescription: {
+                isApplied: ({ editingElement }) => !!editingElement.querySelector(".description"),
+                apply: ({ editingElement }) => {
+                    editingElement.append(renderToElement("html_builder.GoogleMapsDescription"));
+                },
+                clean: ({ editingElement }) => {
+                    editingElement.querySelector(".description").remove();
+                },
+            },
+        };
+    }
+
+    /**
+     * Get the stored API key if any (or open a dialog to ask the user for one),
+     * load and configure the Google Maps API, then use it to set the pin's
+     * initial address.
+     *
+     * @param {Element} editingElement
+     * @param {boolean} [forceReconfigure=false]
+     * @returns {Promise<void>}
+     */
+    async loadGoogleMaps(editingElement, forceReconfigure = false) {
+        /** @type {string  |undefined} */
+        let apiKey = await this.googleMapsService.getGMapsAPIKey();
+        const didReconfigure = await this.configureGMapsAPI({ apiKey, force: forceReconfigure });
+        apiKey = await this.loadGoogleMapsAPIFromService(didReconfigure);
+        if (apiKey) {
+            await this.onGoogleMapsLoaded(editingElement);
+        }
+    }
+
+    /**
+     * Load the Google Maps API from the Google Map Service.
+     * This method is set apart so it can be overridden for testing.
+     *
+     * @param {boolean} [shouldRefetch]
+     * @returns {Promise<string|undefined>} A promise that resolves to an API
+     *                                      key if found.
+     */
+    async loadGoogleMapsAPIFromService(shouldRefetch) {
+        return this.googleMapsService.loadGMapsAPI(true, shouldRefetch);
+    }
+
+    /**
+     * Try to fail early in the case of a misconfiguration by doing a nearby
+     * search on the coordinates set on the editing element. If it works, set
+     * the result's address as the pin's initial address. Otherwise, remove the
+     * snippet (unless the error is being handled).
+     *
+     * @param {Element} editingElement
+     * @returns {Promise<void>}
+     */
+    async onGoogleMapsLoaded(editingElement) {
```

Comment:
I suggest to change the method name that tell what the method is doing? `updateElementMap`?
Actually, I'm not sure what is the method supposed to do as I don't understand why we set pinAdress or remove the element. Reading the doc of the method did not help me understand what it is supposed to do or in which flow it should be called.

Why is it `loadGoogleMaps` calling this method?
It look like we want to update an element that need to load google map if I understand correctly.

```js
async updateElementMap(editingElement, forceReconfigure) {
  const loaded = await this.loadGoogleMaps(editingElement, forceReconfigure)
  if (!loaded) return;
  ...
}
```

Reply:
We set the `pinAddress` so that the input element (google autocomplete) has the name of the place.

The main reason to call `nearbySearch` after loading is to fail early if the API configuration wasn't right. In the original implementation, if I remember correctly, we ended up doing 2 `nearbySearch`es: one to fail early, followed by one to set the `pinAddress` (it was not necessarily obvious because of `trigger_up`s etc.). So I combined the two here, and while I'm at it I'm also setting `this.mapsAPI` and `this.placesAPI` for convenience.

I could have this inside `loadGoogleMaps`, because this only needs to be called once, after loading. Probably better to do that.